src/js/components/widgets-link/widget-link-number.react.js

```diff
@@ -0,0 +1,287 @@
+     */
+    handleClick () {
+        this.setState({ displayPicker: !this.state.displayPicker });
+
+        let widgetlink = document.getElementById('widget-links');
+        ReactDOM.unmountComponentAtNode(widgetlink);
+    }
+
+    /**
+     * Start of a drag event
+     */
+    onMouseDown (e) {
+        this.drag = true; // START of a drag event
+        this.overPoint = true; // Change the look of the point within the canvas
+
+        let mousePos = this._getMousePos(this.refs.canvas, e);
+        this.prevOffset = mousePos.x;
+        this._drawCanvas();
+    }
+
+    /**
+     * While user is dragging
+     */
+    onMouseMove (e) {
+        if (this.drag === true) { // If user is dragging mouse
+            let mousePos = this._getMousePos(this.refs.canvas, e);
+
+            let x = mousePos.x;
+
+            let vel = x - this.prevOffset;
+            let offset = this.offsetX - vel;
+
+            this._setValue(offset / this.center);
+            this.prevOffset = x;
+
+            this._drawCanvas();
+            this._updateEditor(this.value.toFixed(3));
+        }
+    }
+
+    /**
+     * When user stops dragging
+     */
+    onMouseUp () {
+        this.drag = false; // STOP a drag event
+        this.overPoint = false; // Change the look of the point within the canvas
+        this._drawCanvas(); // Draw the new point
+    }
+
+
+    /**
+     * onMouseLeave accounts for the case where the user is still dragging but outside of the widget. The drag event should end.
+     */
+    onMouseLeave () {
+        this.drag = false;
+        this.overPoint = false; // Change the look of the point within the canvas
+        this._drawCanvas(); // Draw the new point
+    }
+
+    /**
+     * Function to get a mouse position within the canvas element
+     */
+    _getMousePos (canvas, evt) {
+        var rect = canvas.getBoundingClientRect();
+        return {
+            x: evt.clientX - rect.left,
+            y: evt.clientY - rect.top
+        };
+    }
+
+    /**
+     * When user srolls wheel
+     * TODO: fine tune this scroll function
+     */
+    onWheel (e) {
+        let x = e.deltaY;
+
+        let vel = x - this.prevWheelOffset;
+        let offset = this.offsetX - vel;
+
+        this._setValue(offset / this.center);
+        // this.prevOffset = x;
+
+        this._drawCanvas();
+        this._updateEditor(this.value.toFixed(3));
+
+        this.prevWheelOffset = e.deltaY - this.prevWheelOffset;
+    }
+
+    /**
+     * Official React lifecycle method
+     * Called every time state or props are changed
+     */
+    render () {
+        return (
+            <Modal id='modal-test' dialogComponentClass={DraggableModal} x={this.x} y={this.y} enforceFocus={false} className='widget-modal' show={this.state.displayPicker} onHide={this.handleClick}>
+                <div className='drag'>
+                    <Button onClick={ this.handleClick } className='widget-exit'><Icon type={'bt-times'} /></Button>
```

Comment:
Would it make sense to use the × (Unicode character) instead of an Icon component here?